src/include/codegen/hash_table.h

```diff
@@ -119,12 +130,23 @@ class HashTable {
                              ProbeCallback &probe_callback,
                              InsertCallback &insert_callback) const = 0;
 
+  // Probe the hash table and insert a new slot if needed, returning both the
+  // result and the data pointer
+  virtual ProbeResult ProbeOrInsert(
+      CodeGen &codegen, llvm::Value *ht_ptr, llvm::Value *hash,
+      const std::vector<codegen::Value> &key) const = 0;
```

Comment:
I briefly remember our discussion on this, but why can't you use the existing `ProbeOrInsert` method?

Reply:
The hash probing introduces a new branch, after which the null bitmap values have to be merged (Phi generation). When building the lang::If for the probing manually, I can pass it to the existing merge function. (https://github.com/tcm-marcel/peloton/blob/c3f7329abdbfc39ae2f941e4029431144cd5ed53/src/codegen/aggregation.cpp#L596 )

Also currently both the SUM and the COUNT part of an AVG aggregation make a hash table lookup, but I wanted to change this in the future. Therefore the lookup result has to be passed on to the next aggregation. (I can't exactly remember why I left this out, I think I will add it just now)